docs/general/ddl/data-types.rst

```diff
@@ -437,6 +443,69 @@ In these expressions, the desired time zone is specified as a string
 The scalar function :ref:`TIMEZONE <scalar-timezone>` (zone, timestamp) is
 equivalent to the SQL-conforming construct timestamp AT TIME ZONE zone.
 
+.. _time-data-type:
+
+time with time zone
+-------------------
+
+The time type consist of time followed by an optional time zone.
+
+``timetz`` is an alias for `time with time zone`.
+
+`time with time zone` literals can be constructed using a string literal
+and a cast. The syntax for string literal is as follows:
+
+.. code-block:: text
+
+    time-element [offset]
+
+    time-element: time-only [fraction]
+    time-only:    HH[[:][mm[:]ss]]
+    fraction:     '.' digit+
+    offset:       {+ | -} time-only | geo-region
+    geo-region:   As defined by ISO 8601.
+
+
+Where `time-only` can contain optional seconds, or optional minutes and seconds,
+and can use `:` as a separator optionally.
+
+`fraction` accepts up to 6 digits, as precision is in micro seconds.
+
+Time zone syntax as defined by `ISO 8601 time zone designators`_.
+
+.. NOTE::
+
+    This type cannot be created in `CREATE TABLE` or `ALTER` statements, as
+    it is meant to be used as parameter and return type for time related
```

Comment:
Not sure about the wording here. Maybe the second sentence could be omitted altogether.